src/Microsoft.AspNet.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -36,8 +36,7 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
                     return false;
                 }
 
-                Context.Response.StatusCode = 500;
-                return true;
+                throw errorContext.Error;
```

Comment:
This could already be a caught exception so re-throwing it will reset the call stack. Do you need to wrap it?